contracts/schemes/Competition.sol

```diff
@@ -0,0 +1,396 @@
+            require(suggestions[topSuggestions[i]].suggester == address(0), "not all winning suggestions redeemed");
+        }
+
+        (, , , , , , ,
+        uint256 nativeTokenRewardLeft,
+        ,
+        uint256 ethRewardLeft,
+        uint256 externalTokenRewardLeft)
+        = ContributionRewardExt(contributionRewardExt).organizationProposals(_proposalId);
+
+        Avatar avatar = ContributionRewardExt(contributionRewardExt).avatar();
+
+        ContributionRewardExt(contributionRewardExt).redeemExternalTokenFromExtContract(
+        _proposalId, address(avatar), externalTokenRewardLeft);
+
+        ContributionRewardExt(contributionRewardExt).redeemEtherFromExtContract(
+        _proposalId, address(avatar), ethRewardLeft);
+
+        ContributionRewardExt(contributionRewardExt).redeemNativeTokenFromExtContract(
+        _proposalId, address(avatar), nativeTokenRewardLeft);
+    }
+
+    /**
+    * @dev getOrderedIndexOfSuggestion return the index of specific suggestion in the winners list.
+    * @param _suggestionId suggestion id
+    */
+    function getOrderedIndexOfSuggestion(uint256 _suggestionId)
+    public
+    view
+    returns(uint256 index) {
+        bytes32 proposalId = suggestions[_suggestionId].proposalId;
+        require(proposalId != bytes32(0), "suggestion does not exist");
+        uint256[] memory topSuggestions = proposals[proposalId].topSuggestions;
+        /** get how many elements are greater than a given element*/
+        for (uint256 i; i < topSuggestions.length; i++) {
+            if (suggestions[topSuggestions[i]].totalVotes > suggestions[_suggestionId].totalVotes) {
+                index++;
+            }
+        }
+    }
+
+    /**
+    * @dev refreshTopSuggestions this function maintain a winners list array.
+    * it will check if the given suggestion is among the top suggestions, and if so,
+    * update the list of top suggestions
+    * @param _proposalId proposal id
+    * @param _suggestionId suggestion id
+    */
+    function refreshTopSuggestions(bytes32 _proposalId, uint256 _suggestionId) private {
+        uint256[] storage topSuggestions = proposals[_proposalId].topSuggestions;
+        if (topSuggestions.length < proposals[_proposalId].numberOfWinners) {
+            topSuggestions.push(_suggestionId);
+        } else {
+         /** get the index of the smallest element **/
+            uint256 smallest = 0;
+            for (uint256 i; i < proposals[_proposalId].numberOfWinners; i++) {
+                if (suggestions[topSuggestions[i]].totalVotes <
+                    suggestions[topSuggestions[smallest]].totalVotes) {
+                    smallest = i;
+                } else if (topSuggestions[i] == _suggestionId) {
+                  //the suggestion is already in the topSuggestions list
+                    return;
+                }
+            }
+
+            if (suggestions[topSuggestions[smallest]].totalVotes < suggestions[_suggestionId].totalVotes) {
```

Comment:
So, if the _suggestionId has the **same** number of votes as the smallest in the topSuggestions, it will **not** be added to the topSuggestions. 

That seems unfair :-)

Reply:
- in this case all the suggestions which has the same number of votes will get their split reward divided by their count.
- for this we are maintaining votesToSuggestions